src/main/java/org/openrewrite/rewrite/CorrectlySpacedDescriptions.java

```diff
@@ -0,0 +1,153 @@
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.rewrite;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.DeclaresMethod;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+
+public class CorrectlySpacedDescriptions extends Recipe {
+
+  private static final String STARTS_AND_ENDS_WITH_NON_WHITESPACE_CHAR = "^\\S.*\\S$";
+  private static final String STARTS_WITH_NON_WHITESPACE_CHAR_ENDS_WITH_SPACE = "^\\S.*\\S $";
+  private static final String STARTS_WITH_NON_WHITESPACE_CHAR_ENDS_WITH_LINEBREAK = "^\\S.*\\s*\\n+\\s*$";
+  private static final String ENDS_WITH_LINEBREAK = "[\\s\\S]*\\n+\\s*$";
+  private static final String IS_ONLY_WHITESPACE = "^\\s*$";
+  private static final String IS_CORRECTLY_SPACED_MAYBE_MD_LIST = "^\\s[-*]\\s\\S*\\n$";
+  private static final String IS_MAYBE_MD_LIST = "^\\s?[-*]\\s\\S[\\s\\S]*";
+  private static final String IS_MAYBE_END_OF_MD_LINK = ".*]$";
+
+  private static final MethodMatcher GET_DESCRIPTION_MATCHER = new MethodMatcher("org.openrewrite.Recipe getDescription()", true);
+
+  @Override
+  public String getDisplayName() {
+    return "Correctly spaced descriptions";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Recipe descriptions should be cleanly formatted. This recipe forces correct spacing in multiline descriptions. " +
+      "In a multi line description the lines should not start with whitespace and end with a single space " +
+      "except for the last line which should end with a \".\" " +
+      "(e.g.\n  ```return \"This is a correct \" + \n   \"multi line description\";```).";
+  }
+
+  @Override
+  public TreeVisitor<?, ExecutionContext> getVisitor() {
+    JavaIsoVisitor<ExecutionContext> visitor = new JavaIsoVisitor<ExecutionContext>() {
+      @Override
+      public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+        if (GET_DESCRIPTION_MATCHER.matches(method.getMethodType())) {
+          J.MethodDeclaration md = (J.MethodDeclaration) new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.Binary visitBinary(J.Binary b, ExecutionContext ctx) {
+              if (isLiteralString(b.getLeft())) {
+                return handle(b).withRight(updateLiteral((J.Literal) b.getRight(), false));
+              } else {
+                return handle(b);
+              }
```

Comment:
I feel we're complicating the logic below a lot by mixing concerns, and not using `super.visitBinary` meaning longer chains of concatenations might not fully be handled in a single iteration.

How would you feel about making the code flow more explicit by adopting something like the following here:
```java
@Override
public J.Binary visitBinary(J.Binary binary, ExecutionContext ctx) {
  J.Binary b = super.visitBinary(binary, ctx);
  b = maybeSuffixLeftWithSpace(b);
  b = maybeSuffixLeftWithNewline(b);
  b = maybePrefixRightWithSpace(b);
  b = maybeSuffixRightWithDot(b);
  return b;
}
```